server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala

```diff
@@ -101,6 +101,8 @@ private[http4s] object Http4sWebSockets {
         case (None, f: WebSocketFrame.Pong)                                  => (None, Some(f))
         case (None, f: WebSocketFrame.Close)                                 => (None, Some(f))
         case (None, f: WebSocketFrame.Data[_]) if f.finalFragment            => (None, Some(f))
+        case (None, f: WebSocketFrame.Text)                                  => (Some(Right(f.payload)), None)
```

Comment:
indeed this seems rather basic :D

Reply:
but if http4s concatenates frames by itself, shouldn't we remove this? it's dead code anyway?

Reply:
Ok, turns out that:
1. Http4s can be configured to not concatenate fragmented frames and not ignore pings/pongs on the level of `WebSocketBuilder2`:
```scala
      .withHttpWebSocketApp(wsb => Router("/" -> wsRoutes(wsb.withDefragment(false).withFilterPingPongs(false)).orNotFound)
```
2. This has no effect if used with Blaze, which always automatically defragments and filters pings pongs
3. It works with Ember though.

I'll remove these stages from `Http4sWebSockets` and add some documentation.

Reply:
Or we can keep our own defragmentation and force `wsb.withDefragment(false)` to make this Tapir-native switch work on Ember by forcing passing through our concatenation :thinking:

Reply:
Ah ... if it's optional, let's keep the code, and let the user of http4s decide. So I guess the code is good as-is

Reply:
It has several bugs actually, as I discovered when I switched from Blaze to Ember in tests. Fixes are on the way in a separate PR then ;)